### PR TITLE
[ll] shader module API rework

### PR DESCRIPTION
**Pull request by kvark**
I made it pretty Vulkan-ic :)
In the future, we may add specialization to `EntryPoint` struct.

### CODE REVIEW

File: src/backend/vulkan/src/device.rs [hunk 1]
```diff
@@ -573,61 +501,49 @@ impl d::Device<B> for Device {
...
-                        Ok(_) => {
-                            let pipeline = *pipelines.next().unwrap();
-
-                            // Check if pipeline compiled correctly
-                            if pipeline == vk::Pipeline::null() {
-                                Err(pso::CreationError::Other) // TODO
-                            } else {
-                                Ok(n::GraphicsPipeline(pipeline))
-                            }
+            Ok(pipelines) |
```
Comment (msiglreith) [inline src/backend/vulkan/src/device.rs]: Nice one! I remember that this logic was a pain to write :)

File: src/backend/vulkan/src/device.rs [hunk 2]
```diff
@@ -711,12 +612,34 @@ impl d::Device<B> for Device {
...
             self.raw.0.create_framebuffer(&info, None)
-                        .expect("error on framebuffer creation")
-        };
+        }.expect("error on framebuffer creation");
 
         n::FrameBuffer { raw: framebuffer }
     }
 
+    fn create_shader_module(&mut self, spirv_data: &[u8]) -> Result<n::ShaderModule, d::ShaderError> {
+        assert_eq!(spirv_data.len() & 3, 0);
```
Comment (msiglreith) [inline src/backend/vulkan/src/device.rs]: Could we add a comment for this. The size of SPIR-V modules requiring to be a multiple of the size of DWORD is not that obvious imo.

File: src/core/src/lib.rs [hunk 3]
```diff
@@ -214,6 +214,34 @@ pub struct HeapType {
     pub heap_index: usize,
 }
 
+/// Shader entry point.
+#[derive(Debug)]
+pub struct EntryPoint<'a, B: Backend> {
```
Comment (msiglreith) [inline src/core/src/lib.rs]: Could be moved into `pso` imo.

**Comment by kvark** [comment]
@msiglreith thanks for reviewing! Everything should be addressed now

**Pull request merged**